editor/src/components/canvas/canvas-strategies/strategies/set-padding-strategy.tsx

```diff
@@ -141,23 +144,78 @@ export const setPaddingStrategy: CanvasStrategyFactory = (canvasState, interacti
 
       const padding = simplePaddingFromMetadata(canvasState.startingMetadata, selectedElement)
       const currentPadding = paddingForEdge(edgePiece, padding)
-      const delta = Math.max(-currentPadding, deltaFromEdge(drag, edgePiece))
+      const rawDelta = deltaFromEdge(drag, edgePiece)
+      const delta = Math.max(-currentPadding, rawDelta)
       const newPadding = offsetPaddingByEdge(
+        edgePiece,
+        rawDelta,
+        padding,
+        precisionFromModifiers(interactionSession.interactionData.modifiers),
+      )
+
+      const newPaddingMaxed = offsetPaddingByEdge(
         edgePiece,
         delta,
         padding,
         precisionFromModifiers(interactionSession.interactionData.modifiers),
       )
-      const paddingString = paddingToPaddingString(newPadding)
 
-      return strategyApplicationResult([
-        ...IndividualPaddingProps.map((p) =>
-          deleteProperties('always', selectedElement, [stylePropPathMappingFn(p, ['style'])]),
-        ),
-        setProperty('always', selectedElement, StylePaddingProp, paddingString),
+      const basicCommands: CanvasCommand[] = [
         updateHighlightedViews('mid-interaction', []),
         setCursorCommand(pickCursorFromEdge(edgePiece)),
         setElementsToRerenderCommand(selectedElements),
+      ]
+
+      if (rawDelta < PaddingTearThreshold) {
+        const paddingPropToBeRemoved = paddingPropForEdge(edgePiece)
+        return strategyApplicationResult([
+          ...basicCommands,
+          deleteProperties('always', selectedElement, [
+            StylePaddingProp,
+            stylePropPathMappingFn(paddingPropToBeRemoved, ['style']),
+          ]),
+          ...IndividualPaddingProps.filter(
```

Comment:
Could this be broken out into a variable? I think with some sensible name it would make the purpose of this a little clearer.

Reply:
thanks, done!

Reply:
done!